File: LeetCode-Python3/1. Two Sum/1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> mp;
        for (int i = 0; i < nums.size(); i++)
            mp[nums[i]] = i;
        vector<int> ans;
        for (int i = 0; i < nums.size(); i++)
        {
            int t = target - nums[i];
            if (mp.count(t) && i != mp[t])
            {
                ans.push_back(i);
                ans.push_back(mp[t]);
                return ans;
            }
        }
        
    }
};

File: LeetCode-Python3/1. Two Sum/1.py
# -*- coding: utf-8 -*-
"""
Created on Thu Sep 20 13:06:43 2018

@author: Administrator
"""

class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        seen = {}
        for i, num in enumerate(nums):
            want = target - num
            if want in seen:
                return [seen[want], i]
            seen[num] = i
        return []
                
